### PR TITLE
Scrum 1438 final

### PR DESCRIPTION
**Pull request by ianlongden**

### CODE REVIEW

File: agr_literature_service/api/crud/reference_crud.py [hunk 1]
```diff
@@ -409,3 +416,92 @@ def merge_references(db: Session,
...
+               journal, volume, issue, page_range)
+    return citation
+
+
+def author_order_sort(author: AuthorModel):
+    return author.order
+
+
+def citation_from_data(reference_data, authorNames):
+    authorNames = authorNames[:-2]  # remove last '; '
```
Comment (valearna) [inline agr_literature_service/api/crud/reference_crud.py:435]: Can we remove the last '; ' in the main crud function or leave it here but check if `authorNames.endswith('; ')`?

**Review by valearna — APPROVED** [review]
This looks great, thanks @ianlongden

File: agr_literature_service/api/crud/reference_crud.py [hunk 2]
```diff
@@ -108,6 +110,8 @@ def create(db: Session, reference: ReferenceSchemaPost):  # noqa
                         obj_data["orcid_cross_reference"] = cross_reference_obj
                     del obj_data["orcid"]
                     db_obj = create_obj(db, AuthorModel, obj_data, non_fatal=True)
+                    if db_obj.name:
+                        authorNames += db_obj.name + '; '
```
Comment (azurebrd) [inline agr_literature_service/api/crud/reference_crud.py:114]: Are these authors at this point ordered by the 'order' field ?   I see that happens in get_citation_from_obj but not in get_citation_from_args, which uses authorNames from here ?

Comment (azurebrd) [reply agr_literature_service/api/crud/reference_crud.py:114]: Oh, this is only when creating a reference, so the authors are already ordered.

File: agr_literature_service/api/crud/reference_crud.py [hunk 3]
```diff
@@ -409,3 +416,92 @@ def merge_references(db: Session,
...
+def get_citation_from_args(authorNames: str, year: str, title: str, journal: str,
+                           volume: str, issue: str, page_range: str):
+    # Create the citation from the args given.
+    citation = "{}, ({}) {} {} {} ({}): {}".\
+        format(authorNames, year, title,
+               journal, volume, issue, page_range)
+    return citation
+
+
+def author_order_sort(author: AuthorModel):
```
Comment (azurebrd) [inline agr_literature_service/api/crud/reference_crud.py:430]: This isn't being used ?

File: alembic/scripts/add_citation.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    print("Starting {}".format(datetime.datetime.now()))
+    outfile = open("citation_update.txt", 'a')
+    seen = {}
+    i = 0
+    last_seen = 0
+    batch_size = 1000
+
+    descending = db.query(ReferenceModel).order_by(ReferenceModel.reference_id.desc())
+    last_ref = descending.first()
+    print("Last referecne_id is : {}".format(last_ref.reference_id))
```
Comment (azurebrd) [inline alembic/scripts/add_citation.py:42]: referecne_id -> reference_id 
not important

**Review by azurebrd — APPROVED** [review]
Thanks Ian

File: alembic/scripts/add_citation.py (hunk 4 above)
Comment (ianlongden) [reply alembic/scripts/add_citation.py:42]: I will fix typos etc later.

File: agr_literature_service/api/crud/reference_crud.py (hunk 1 above)
Comment (ianlongden) [reply agr_literature_service/api/crud/reference_crud.py:435]: I will revisit when I do more work. 
But yes that makes sense.

File: agr_literature_service/api/crud/reference_crud.py (hunk 3 above)
Comment (ianlongden) [reply agr_literature_service/api/crud/reference_crud.py:430]: Yep, I changed the way I did things in the end. Will Remove later.

**Pull request merged**